### ISSUE TITLE
Refactor container cleanup

### ISSUE THREAD

**Issue by stoically** [stoically/temporary-containers#305]
Under some rare circumstances the queue to remove containers can get stuck, so a Firefox restart is necessary to remove containers. Refactoring the container removal code to not be a total mess should help with that.

**Comment by stoically** [comment]
@iKarith The "stuck" scenario *should* not influence Firefox performance, since all that happens is that TCs are not immediately cleaned up. What you describe sounds like a different problem, but to be sure I'd need a [debug log](https://github.com/stoically/temporary-containers/wiki/Debug-Log#while-the-add-on-initializes) while the Add-on initializes - feel free to comment it here, or open a new issue, whatever you prefer, thanks!

**Comment by iKarith** [comment]
Oh, the performance issue was caused by Facebook, not by temporary containers. The issue was after a browser restart, sometimes temporary containers don't get reaped, even though there are no tabs using them. They basically hang around until you delete them manually.

I generally worked around the problem by reducing the reap time from 15 minutes to 5 minutes. If I accidentally close something I didn't mean to, I usually know about it instantly.

**Comment by stoically** [comment]
First round of container cleanup is done, something that still needs to be done:

[ ] Instead of hammering the API with "does the container still have open tabs" checks for every container, it should do that only once when the queue begins trying to remove containers

**Comment by stoically** [comment]
@iKarith 

> The issue was after a browser restart, sometimes temporary containers don't get reaped, even though there are no tabs using them. They basically hang around until you delete them manually.

That's interesting. Do you by chance have an `about:sessionrestore` tab open when that happens?

Other than that it might be useful to have a way to manually start removing containers, maybe on the statistics tab. I'll look into that if I get a chance.
